### PR TITLE
EOS-19426: Check swap size <= 16T. If its > 16T, create seperate swap…

### PR DESCRIPTION
**Pull request by atulsdeshmukh2312**
… disks of max 16T each

Signed-off-by: Atul Deshmukh <atul.deshmukh@seagate.com>

### CODE REVIEW

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
 import os
 import re
 import subprocess
+from decimal import Decimal
```
Comment (huanghua78) [inline scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:25]: Please fix the Codacy warnings here.

'decimal.Decimal' imported but unused (F401)

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 2]
```diff
@@ -246,6 +251,44 @@ def create_swap(self, swap_dev):
...
     add_swap_fstab(self, swap_dev)
 
+'''
+    Create swap disks of VG.
+    If swap size is more than 16T then
+    create more than one swap disks each of max 16T.
+'''
+def create_swap_disks(self, index, swap_sz, vg_name):
+    disk_count = 1
+    while(swap_sz > MAX_SWAP_DISK_SZ):
```
Comment (huanghua78) [inline scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:261]: add a space after 'while'
```
        while (swap_sz > MAX_SWAP_DISK_SZ):
```

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 3]
```diff
@@ -246,6 +251,44 @@ def create_swap(self, swap_dev):
...
+        execute_command(self, cmd)
+        disk_count += 1
+        swap_dev = f"/dev/{vg_name}/{lv_swap_name}"
+        create_swap_entries(self, swap_dev)
+    lv_swap_name = f"lv_main_swap{index}_{disk_count}"
+    swap_dev = f"/dev/{vg_name}/{lv_swap_name}"
+    swap_sz = round(swap_sz, 3)
+    cmd = f"lvcreate -n {lv_swap_name} {vg_name} -L {swap_sz}T --yes"
+    execute_command(self, cmd)
+    create_swap_entries(self, swap_dev)
```
Comment (huanghua78) [inline scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:275]: I know this is to handle the last one which is less than "MAX_SWAP_DISK_SZ".

But I think we can avoid duplicating the code.
Can we do this in the loop?

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 4]
```diff
@@ -521,30 +552,56 @@ def get_metadata_disks_count(self):
         for device in metadata_devices:
             dev_count += 1
     return dev_count
+'''
```
Comment (huanghua78) [inline scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:555]: Leave at least one blank line between functions.

**Comment by huanghua78** [comment]
Hello @atulsdeshmukh2312 please update the status of this PR.

**Comment by andriytk** [comment]
@atulsdeshmukh2312, please edit the summary line and description of this PR. Make the summary line shorter (try to make it no longer than 50 chars), describe details in the description.

**Pull request closed without merging**